### PR TITLE
add some email validation

### LINKED ISSUE #1207: Confirmation email to corresponding author possibly sent multiple times
**Describe the bug**
Every (auto) saving of the manuscript seems to send a new email to the address entered in the first step of the initial submission.

**Steps To Reproduce**
1. Start an initial submission
2. Enter first name, last name, email address and institution
3. Wait a few seconds
4. Read the log line `info: Sent confirmation email to corresponding author: g.sironi@elifesciences.org` written multiple times

**Expected behavior**
If https://github.com/elifesciences/elife-xpub/pull/1206 is correct this means an email is going out on each autosave of the form.

**Screenshots**
Some more logs I have inserted:
```
info: manuscript.getAuthor(): null
info: newAuthorEmails: g.sironi@elife
info: originalAuthorEmails: 
debug: Updated manuscript manuscriptId=7270b619-fd16-4fae-8771-18fd64efdcb6, userId=75c11a2d-9185-4b6a-a68d-86a7f60eae64
info: POST /graphql 200 UpdateManuscript 1410 218.026 ms

info: Sent confirmation email to corresponding author: g.sironi@elife
info: manuscript.getAuthor(): [object Object]
info: newAuthorEmails: g.sironi@elifesciences.org
info: originalAuthorEmails: [object Object]
debug: Updated manuscript manuscriptId=7270b619-fd16-4fae-8771-18fd64efdcb6, userId=75c11a2d-9185-4b6a-a68d-86a7f60eae64
info: POST /graphql 200 UpdateManuscript 1428 149.295 ms

info: Sent confirmation email to corresponding author: g.sironi@elifesciences.org
```
It appears that there is a comparison between `[object Object]` and actual strings around https://github.com/elifesciences/elife-xpub/blob/develop/server/xpub-server/entities/manuscript/resolvers/updateManuscript.js#L24

Moreover, the autosave (if there is one) may be firing too soon saving partial results like `g.sironi@elife` rather than `g.sironi@elifesciences.org`.

### PR DESCRIPTION
**Pull request by diversemix**
#### Background

Before checking if the email has changed, properly validate the email before sending. 

#### Any relevant tickets

Close #1207

#### How has this been tested?

Locally by looking at the logs

### CODE REVIEW

File: server/xpub-server/entities/manuscript/resolvers/updateManuscript.js [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+const isValidEmail = async email => {
+  let result = false
+  try {
+    await Joi.validate(
+      email,
+      Joi.string()
+        .email()
+        .required(),
+    )
+    const tld = email.split('@')[1]
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: We do not get to this point if we do not have an "@" char in the email field

File: server/xpub-server/entities/manuscript/resolvers/updateManuscript.js [hunk 2]
```diff
@@ -1,10 +1,33 @@
...
+    const lookup = await resolveMx(tld)
```
Comment (giorgiosironi) [inline server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: taking the opportunity to ask whether all the `await` constructs will result in a blocked process (iowait) or the single Node process will switch to some other thing from the event queue? I think the latter but better to get confirmation given all the network calls going on

Comment (giorgiosironi) [reply server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: separately: does this need to be stubbed in the context of unit tests (no network, run on a train) or disabled via configuration?

Comment (damnhipster) [reply server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: @giorgiosironi Node will start process something else from the event queue during an `await`

File: server/xpub-server/entities/manuscript/resolvers/updateManuscript.js [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+const util = require('util')
 const config = require('config')
 const mailer = require('@pubsweet/component-send-email')
 const logger = require('@pubsweet/logger')
-
 const { Manuscript, User } = require('@elifesciences/xpub-model')
 
+const resolveMx = util.promisify(dns.resolveMx)
+
+const isValidEmail = async email => {
```
Comment (damnhipster) [inline server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: We can move this function to the controller to start with, and stub it out for testing purposes.

File: server/xpub-server/entities/manuscript/resolvers/updateManuscript.js (hunk 2 above)
Comment (diversemix) [reply server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: @giorgiosironi / @damnhipster .. the notification is now done asynchronously with the new commit

File: server/xpub-server/entities/manuscript/resolvers/updateManuscript.js (hunk 3 above)
Comment (diversemix) [reply server/xpub-server/entities/manuscript/resolvers/updateManuscript.js]: Yup - sorted in the new commit.

File: server/xpub-controller/notification.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+const logger = require('@pubsweet/logger')
+
+class Notification {
+  constructor(config, people) {
+    this.resolveMx = util.promisify(dns.resolveMx)
+    this.people = people
+    this.config = config
+  }
+
+  async isValidEmail(email) {
```
Comment (diversemix) [inline server/xpub-controller/notification.js]: This should perhaps be a static method on the class

File: server/xpub-controller/notification.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  async sendDashboardEmail() {
+    const emailList = this.people.map(person => person.alias.email).join(',')
+    const valid = await this.people
+      .map(async person => this.isValidEmail(person.alias.email))
+      .reduce((previous, current) => previous && current)
+
+    if (!valid) {
+      return false
```
Comment (damnhipster) [inline server/xpub-controller/notification.js]: It's worth logging here, if only so that we can manually test it in the review environment.

**Pull request merged**